test/testdrive/source-format-json.td

```diff
@@ -0,0 +1,50 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Verify behavior of FORMAT JSON
+
+$ kafka-create-topic topic=data partitions=1
+$ kafka-ingest format=bytes topic=data
+{"a":"b","c":"d"}
+
+> CREATE CONNECTION kafka_conn
+  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+> CREATE SOURCE data
+  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-data-${testdrive.seed}')
+  FORMAT JSON;
```

Comment:
Can one have `KEY FORMAT JSON`? ` or `ENVELOPE UPSERT`  over `FORMAT JSON` ? If yes, would you mind adding a quick test for this?

Reply:
Added these tests